### PR TITLE
Add changes from 2457 to Timex unit tests

### PR DESCRIPTION
**Pull request by FedericoBernal**

### CODE REVIEW

File: Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java [hunk 1]
```diff
@@ -69,6 +69,177 @@ public void dataTypesResolverDateSunday()
...
+        Assert.assertEquals("date", resolution.getValues().get(1).getType());
+        Assert.assertEquals("2019-05-06", resolution.getValues().get(1).getValue());
+        Assert.assertNull(resolution.getValues().get(1).getStart());
+        Assert.assertNull(resolution.getValues().get(1).getEnd());
+    }
+
+    @Test
+    public void DataTypes_Resolver_Date_Feb_2nd()
+    {
+        LocalDateTime today = LocalDateTime.of(2020, 10, 20, 0, 0);
```
Comment (Batta32) [inline Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java]:
```suggestion
        LocalDateTime today = LocalDateTime.of(2020, 10, 20, 0, 0, 0);
```

File: Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java [hunk 2]
```diff
@@ -69,6 +69,177 @@ public void dataTypesResolverDateSunday()
...
+        Assert.assertEquals("date", resolution.getValues().get(1).getType());
+        Assert.assertEquals("2021-02-02", resolution.getValues().get(1).getValue());
+        Assert.assertNull(resolution.getValues().get(1).getStart());
+        Assert.assertNull(resolution.getValues().get(1).getEnd());
+    }
+
+    @Test
+    public void DataTypes_Resolver_DateTimeRange_Oct_25th_Afternoon()
+    {
+        LocalDateTime today = LocalDateTime.of(2020, 10, 20,0,0);
```
Comment (Batta32) [inline Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java]:
```suggestion
        LocalDateTime today = LocalDateTime.of(2020, 10, 20, 0, 0, 0);
```

File: Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java [hunk 3]
```diff
@@ -69,6 +69,177 @@ public void dataTypesResolverDateSunday()
...
+        Assert.assertEquals("datetimerange", resolution.getValues().get(1).getType());
+        Assert.assertEquals("2020-10-25 12:00:00", resolution.getValues().get(1).getStart());
+        Assert.assertEquals("2020-10-25 16:00:00", resolution.getValues().get(1).getEnd());
+        Assert.assertNull(resolution.getValues().get(1).getValue());
+    }
+
+    @Test
+    public void DataTypes_Resolver_DateTimeRange_Week11_Monday()
+    {
+        LocalDateTime today = LocalDateTime.of(2020, 10, 20,0,0);
```
Comment (Batta32) [inline Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java]:
```suggestion
        LocalDateTime today = LocalDateTime.of(2020, 10, 20, 0, 0, 0);
```

File: Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java [hunk 4]
```diff
@@ -69,6 +69,177 @@ public void dataTypesResolverDateSunday()
...
+        Assert.assertNull(resolution.getValues().get(1).getStart());
+        Assert.assertNull(resolution.getValues().get(1).getEnd());
+        Assert.assertEquals("2021-03-15", resolution.getValues().get(1).getValue());
+    }
+
+    @Test
+    public void DataTypes_Resolver_DateTimeRange_Thanksgiving()
+    {
+        // XXXX-11-WXX-4-4 -> 4th Thursday (4th ISO weekday) in unspecified week in November in unspecified year
+        LocalDateTime today = LocalDateTime.of(2020, 10, 20, 0, 0);
```
Comment (Batta32) [inline Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java]:
```suggestion
        LocalDateTime today = LocalDateTime.of(2020, 10, 20, 0, 0, 0);
```

File: Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java [hunk 5]
```diff
@@ -69,6 +69,177 @@ public void dataTypesResolverDateSunday()
...
+        Assert.assertEquals("2019-04-W01", resolution.getValues().get(0).getTimex());
+        Assert.assertEquals("daterange", resolution.getValues().get(0).getType());
+        Assert.assertEquals("2019-04-01", resolution.getValues().get(0).getStart());
+        Assert.assertEquals("2019-04-08", resolution.getValues().get(0).getEnd());
+    }
+
+    @Test
+    public void DataTypes_Resolver_DateRange_first_week_of_April()
+    {
+        LocalDateTime today = LocalDateTime.of(2021, 1, 22, 0, 0);
```
Comment (Batta32) [inline Java/libraries/recognizers-text-datatypes-timex-expression/src/test/java/com/microsoft/recognizers/text/datatypes/timex/expression/TestTimexResolver.java]:
```suggestion
        LocalDateTime today = LocalDateTime.of(2021, 1, 22, 0, 0, 0);
```

**Review by Batta32 — CHANGES_REQUESTED** [review]
All the methods in Java should be lowerCamelCase.

Example: `DataTypes_Resolver_Date_6th` -> `dataTypesResolverDate6th`

**Review by fran893 — APPROVED** [review]

**Review by Batta32 — APPROVED** [review]
LGTM

**Pull request merged**